server/src/main/java/com/objectcomputing/checkins/services/kudos/KudosServicesImpl.java

```diff
@@ -269,4 +283,29 @@ private KudosResponseDTO constructKudosResponseDTO(Kudos kudos) {
 
         return kudosResponseDTO;
     }
+
+    void sendNotification(Kudos kudos) {
+        // Only public kudos really need approval, but just in case...
+        if (kudos.getPubliclyVisible()) {
```

Comment:
It looks like this is sending if publicly visible. There needs to be an additional check for publicly visible kudos that confirms that there is an approval date as well. We only want to send public kudos after they have been approved by an admin.

Additionally, if the kudos is not publicly visible, then the notification should be sent.  So, I think this if should really be:

```java
if (!kudos.getPubliclyVisible() || kudos.getDateApproved() != null) {
```

Reply:
This method is only called after the Kudos has been approved.  At this point, the Kudos is certain to have an approved date, as far as I can tell.

Private kudos are handled elsewhere.